src/vs/workbench/contrib/extensions/electron-browser/extensionTipsService.ts

```diff
@@ -1132,5 +1132,87 @@ export class ExtensionTipsService extends Disposable implements IExtensionTipsSe
 			.map(extensionId => (<IExtensionRecommendation>{ extensionId, sources: ['application'] })));
 	}
 
+	// called when Visualizer icon is clicked in the Query Results grid; "Never Show Again" is not an option
+	promptVisualizerExtensions(): void {
+		const storageKey = 'extensionsAssistant/VisualizerRecommendationsIgnore';
+
+		if (this.storageService.getBoolean(storageKey, StorageScope.GLOBAL, false)) {
+			return;
+		}
+
+		let recommendations: IExtensionRecommendation[];
+		let localExtensions: ILocalExtension[];
+		const getRecommendationPromise = this.getVisualizerExtensions().then(recs => { recommendations = recs; });
+		const getLocalExtensionPromise = this.extensionsService.getInstalled(ExtensionType.User).then(local => { localExtensions = local; });
+		Promise.all([getRecommendationPromise, getLocalExtensionPromise]).then(() => {
+			if (!recommendations.every(rec => { return localExtensions.findIndex(local => local.identifier.id.toLocaleLowerCase() === rec.extensionId.toLocaleLowerCase()) !== -1; })) {
+				return new Promise<void>(c => {
+					this.notificationService.prompt(
+						Severity.Info,
+						localize('visualizer.VisualizerExtensions', "To utilize the Visualizer feature, the SandDance extension is required to be installed."),
+						[{
+							label: localize('visualizer.installAll', "Install Extension"),
```

Comment:
Would they really want to install all? I'm not sure we need this - let's discuss in the meeting.

Reply:
Do the problem is you're not doing a specific extension anymore, it's a category. @angdesign and @alanrenmsft can you help here?

Reply:
Latest change: 
**Azure Data Studio has data visualization extension recommendations.**